src/validation.h

```diff
@@ -1038,7 +1038,10 @@ class ChainstateManager
     }
 
 
-    /** Best header we've seen so far (used for getheaders queries' starting points). */
+    /** Best header we've seen so far for which the block is not known to be invalid
+        (used, among others, for getheaders queries' starting points).
+        In case of multiple best headers with the same work, it could point to any:
+        CBlockIndexWorkComparator tiebreaker rules don't apply here. */
```

Comment:
In commit "doc: Improve m_best_header documentation" (fdcdc2cdeadb5d36076f1b94b54261e22e031354)

It seems like it would be nice to use CBlockIndexWorkComparator consistently for comparisons, but since it's probably beyond scope of this PR, it's good to have this documentation comment mentioning the lack of consistency.

Just in case its useful for future reference, the post describing the current situation in the most detail is https://github.com/bitcoin/bitcoin/pull/31405#discussion_r2129780775